javascript/nodejs/header_injection.js

```diff
@@ -0,0 +1,31 @@
+var server = http.createServer(function (req, res) {
+    var bla = 'dsdsd';
+    switch (testIndex++) {
+        case 0:
+            // ruleid: generic_header_injection
+            res.writeHead(200, { test: 'foo \r\ninvalid: bar' + req.foo });
+            break;
+        case 1:
+            // ruleid: generic_header_injection
+            res.writeHead(200, { test: req.foo + 'foo \ninvalid: bar' });
+            break;
+        case 2:
+            // ruleid: generic_header_injection
+            res.writeHead(200, { test: 'foo \rinvalid: bar' + req.foo + 'asdadasd', foo: bar });
+            break;
+        case 3:
+            // sgrep can't detect this!
+            // ruleid: generic_header_injection
```

Comment:
```suggestion
            // todoruleid: generic_header_injection
```

Reply:
Perfect! This is so useful to match string concatenations.